arch/arm/arm32/hypercall.S

```diff
@@ -0,0 +1,45 @@
+#include <xtf/asm_macros.h>
+#include <xen/xen.h>
+
+#define __HVC(imm16) .long \
+    ((0xE1400070 | (((imm16) & 0xFFF0) << 4) | ((imm16) & 0x000F)) & 0xFFFFFFFF)
+
+#define HYPERCALL_SIMPLE(hypercall)     \
```

Comment:
Both .S was borrowed from Linux right? If so, that's fine because the license is not GPL. But please retain the copyright from the author.